packages/shell-api/src/result.ts

```diff
@@ -127,3 +127,21 @@ export class CursorIterationResult extends ShellApiValueClass {
     this.documents = [];
   }
 }
+
+export type GetShardDistributionResult = {
+  Totals: {
+    data: string;
+    docs: number;
+    chunks: number;
+  } & {
+    [individualShardDistribution: string]: string[];
+  };
+} & {
+  [individualShardResult: string]: {
```

Comment:
Optional, feel free to take it or leave it:

```suggestion
    [individualShardDistribution: `Shard ${string} at ${string}`]: string[];
  };
  [individualShardResult: `Shard ${string}`]: {
```

Using template literals we can both be a bit more precise here and avoid the potential conflicts between `Totals`, `data`, `docs`, `chunks` and the generated keys

Reply:
That's awesome! I was thinking of check in this direction but didn't think template literals were that advanced in TS, super cool!